File: app/src/main/java/com/example/jetpackcomponentsapp/model/CustomModel.java
package com.example.jetpackcomponentsapp.model;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class CustomModel {

    public final String id;
    public String name;
    @Nullable public final String icon;
    @Nullable public String file;

    public CustomModel(String name) {
        this.id = "Nil";
        this.name = name;
        this.icon = null;
    }

    public CustomModel(@Nullable String id, @Nullable String name, @Nullable String icon, @Nullable String file) {
        this.id = id;
        this.name = name;
        this.icon = icon;
        this.file = file;
    }

    @NonNull
    @Override
    public String toString() {
        return "CustomModel id " + this.id + ", name " + this.name + ", icon " + this.icon;
    }
}

File: app/src/main/java/com/example/jetpackcomponentsapp/model/CustomModel.kt
package com.example.jetpackcomponentsapp.model

import com.example.jetpackcomponentsapp.R

public class CustomModel {

    companion object {
        private val TAG = CustomModel::class.java.getSimpleName()
    }

    val id : Int?
    var name : String? = null
    val icon : Int

    constructor(name : String) {
        this.id = null
        this.name = name
        this.icon = R.drawable.ic_android_black
    }

    constructor(id : Int, name : String) {
        this.id = id
        this.name = name
        this.icon = R.drawable.ic_android_black
    }

    constructor(id : Int?, name : String, icon : Int?) {
        this.id = id
        this.name = name
        this.icon = icon ?: R.drawable.ic_android_black
    }

    override fun toString(): String {
        return "$TAG($id, $name, $icon)" ?: super.toString()
    }
}